search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQARequestProcessor.java

```diff
@@ -61,9 +75,18 @@ public SearchRequestProcessor create(
             Map<String, Object> config,
             PipelineContext pipelineContext
         ) throws Exception {
-            return new GenerativeQARequestProcessor(tag, description, ignoreFailure,
-                ConfigurationUtils.readStringProperty(GenerativeQAProcessorConstants.REQUEST_PROCESSOR_TYPE, tag, config, GenerativeQAProcessorConstants.CONFIG_NAME_MODEL_ID)
-            );
+            if (featureFlagSupplier.getAsBoolean()) {
+                return new GenerativeQARequestProcessor(tag, description, ignoreFailure,
+                    ConfigurationUtils.readStringProperty(GenerativeQAProcessorConstants.REQUEST_PROCESSOR_TYPE,
+                        tag,
+                        config,
+                        GenerativeQAProcessorConstants.CONFIG_NAME_MODEL_ID
+                    ),
+                    this.featureFlagSupplier
+                );
+            } else {
+                throw new MLException(GenerativeQAProcessorConstants.FEATURE_NOT_ENABLED_ERROR_MSG);
```

Comment:
should this be an MLException or a runtime exception? I think its runtime exception. 

what is your thought?

Reply:
MLException extends RuntimeException.  I chose MLException simply because this error will be coming from ml-commons.  If there was an exception specific to feature flags, that'd be preferable, but there isn't one and I didn't want to introduce a new one in this PR.